python/cugraph/tests/test_random_walks.py

```diff
@@ -0,0 +1,154 @@
+        The maximum depth of the random walks
+
+
+    Returns
+    -------
+    random_walks_edge_lists : cudf.DataFrame
+        GPU data frame containing all random walks sources identifiers,
+        destination identifiers, edge weights
+
+    seeds_offsets: cudf.Series
+        Series containing the starting offset in the returned edge list
+        for each vertex in start_vertices.
+    """
+    G = utils.generate_cugraph_graph_from_file(
+        graph_file, directed=directed, edgevals=True)
+    assert G is not None
+
+    k = random.randint(1, 10)
+    start_vertices = random.sample(range(G.number_of_vertices()), k)
+    df, offsets = cugraph.random_walks(G, start_vertices, max_depth)
+
+    return df, offsets, start_vertices
+
+
+def check_random_walks(df, offsets, seeds, df_G=None):
+    invalid_edge = 0
+    invalid_seeds = 0
+    invalid_weight = 0
+    offsets_idx = 0
+    for i in range(len(df.index)):
+        src, dst, weight = df.iloc[i].to_array()
+        if i == offsets[offsets_idx]:
+            if df['src'].iloc[i] != seeds[offsets_idx]:
+                invalid_seeds += 1
+                print(
+                        "[ERR] Invalid seed: "
+                        " src {} != src {}"
+                        .format(df['src'].iloc[i], offsets[offsets_idx])
+                    )
+            offsets_idx += 1
+
+        edge = df.loc[(df['src'] == (src)) & (df['dst'] == (dst))].reset_index(
+            drop=True)
+        exp_edge = df_G.loc[
+            (df_G['src'] == (src)) & (
+                df_G['dst'] == (dst))].reset_index(drop=True)
+
+        if not exp_edge.equals(edge[:1]):
+            print(
+                    "[ERR] Invalid edge: "
+                    "There is no edge src {} dst {} weight {}"
+                    .format(src, dst, weight)
+                )
+            invalid_weight += 1
+
+    assert invalid_edge == 0
+    assert invalid_seeds == 0
+    assert invalid_weight == 0
+
+# =============================================================================
+# Pytest Setup / Teardown - called for each test function
+# =============================================================================
+
+
+def prepare_test():
+    gc.collect()
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_SMALL)
+@pytest.mark.parametrize("directed", DIRECTED_GRAPH_OPTIONS)
+@pytest.mark.parametrize("max_depth", [None])
+def test_random_walks_invalid_max_dept(
+    graph_file,
+    directed,
+    max_depth
+):
+    """Test calls random_walks an invalid type"""
+    prepare_test()
+    with pytest.raises(TypeError):
+        df, offsets, seeds = calc_random_walks(
+            graph_file,
+            directed=directed,
+            max_depth=max_depth
+        )
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_SMALL)
+@pytest.mark.parametrize("directed", DIRECTED_GRAPH_OPTIONS)
+def test_random_walks(
+    graph_file,
+    directed
+):
+    max_depth = random.randint(2, 10)
```

Comment:
If the test fails, will the user know what randomly chosen `max_depth` was used to get the results? This may need to be printed somewhere too so devs can reproduce the error if necessary.